modules/library/main/src/main/java/org/geotools/styling/visitor/RescalingMode.java

```diff
@@ -76,7 +63,11 @@ public String rescaleToStringInternal(double scaleFactor, Measure measure) {
         @Override
         public String rescaleToStringInternal(double scaleFactor, Measure measure) {
             if (measure.isRealWorldUnit()) {
-                return String.valueOf(measure.value);
+                if(measure.isRealWorldUnitInPixelDefault()){                    
+                    return String.valueOf(measure.value) + UNIT_SYMBOLS.get(measure.uom);
+                }else{
+                    return String.valueOf(measure.value);
```

Comment:
should we explicitly provide a unit of "px" here - since we are making a judgement call